modules/core/CompositeGeometry.h

```diff
@@ -0,0 +1,42 @@
+#pragma once
+
+#include "IPrim.h"
+#include "Solid.h"
+
+namespace rt {
+    enum OperationType {
+        Intersection, Subtraction, Union
+    };
+
+    class CCompositeGeometry : public IPrim {
+    public:
+        DllExport explicit CCompositeGeometry(ptr_shader_t pShader, const CSolid &s1, const CSolid &s2,
+                                              OperationType operationType);
+
+        DllExport explicit CCompositeGeometry(const rt::ptr_shader_t &pShader,
+                                              const rt::CSolid &s1, const rt::ptr_prim_t &p2,
+                                              rt::OperationType operationType);
+
+        DllExport explicit CCompositeGeometry(const ptr_shader_t &pShader, const ptr_prim_t &p1, const ptr_prim_t &p2,
+                                              OperationType operationType);
+
+        DllExport bool intersect(Ray &ray) const override;
+
+        DllExport bool if_intersect(const Ray &ray) const override;
+
+        DllExport void transform(const Mat &T) override;
+
+        DllExport Vec3f getNormal(const Ray &ray) const override;
+
+        DllExport Vec2f getTextureCoords(const Ray &ray) const override;
+
+        DllExport CBoundingBox getBoundingBox() const override;
+
+    private:
+        CSolid m_s1;
```

Comment:
Maybe it make sense to store them as vectors of primitives. Solid has only m_pivot in addition to that container. If m_pivot is needed?

Reply:
I was just thinking that if we wanted to add transformations to composite geometries then it would be easier to use the built in logic of the Solid class. This makes it simpler to just pull all the logic that's existent in this class to the CompositeGeometry class.